config/config.go

```diff
@@ -84,6 +85,23 @@ func (c *Config) SetURL(u string) error {
 		return err
 	}
 
+	// Expand tilde to the users home directory
+	// This is needed in case the shell is unable to expand the path to the users
+	// home directory for inputs like these:
+	// crypto://password@~/path/to/config
+	// NOTE: - url.Parse() will interpret `~` as the Host Element of the URL in
+	//         this case.
+	//       - Using filepath.Abs() on it won't work as it will interpret `~` as
+	//         the name of a regular folder.
+	if url.Host == "~" {
+		usr, err := user.Current()
+		if err != nil {
+			return err
+		}
+
+		url.Path = filepath.Join(usr.HomeDir, url.Path)
```

Comment:
Maybe I'm also missing something, but don't we need to set `url.Host` to an empty string?

Reply:
presumably better to do so :+1: